### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by amosr**
A lot of the documentation is somewhat old and outdated now

### CODE REVIEW

File: doc/internals/todo.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Folds over groups
+-----------------
+
+Groups must end in folds, but there is currently no way to fold over all the values in the group.
+
+Suppose we want to group by days, then find the day with the maximum number of entries:
+```
+feature feat
+~> group fold (key|value) = (group date ~> count)
+~> maximum value
```
Comment (amosr) [inline doc/internals/todo.md]: thoughts for better syntax for this?

Comment (charleso) [inline doc/internals/todo.md]: > thoughts for better syntax for this?

Could we use the `match` syntax somehow if `key|value` is a tuple?

Comment (amosr) [inline doc/internals/todo.md]: prabobably it should be (key,value) instead.
do you mean
```
feature feat
~> match group date ~> count
    | (key,value) -> maximum value
    end
```
?

**Comment by amosr** [comment]
OK to merge?

**Comment by amosr** [comment]
no! tests failing because of the *ugly* output. I might as well fix that now

File: doc/internals/todo.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+Todo for Source
+===============
+
+Case expressions and ifs
+------------------------
+
+We probably need slightly clunkier syntax than Haskell, unless we want to implement full-blown whitespace sensitivity.
+I think something like this would be fine.
```
Comment (jystic) [inline doc/internals/todo.md]: fwiw, the Idris parser is [whitespace sensitive](https://github.com/idris-lang/Idris-dev/blob/master/src/Idris/ParseHelpers.hs#L442) and uses kmett's trifecta parser combinators

File: doc/internals/todo.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+```
+option d v
+ = match v
+   | None   -> d
+   | Some a -> a
+   end
+```
```
Comment (jystic) [inline doc/internals/todo.md]: this looks pretty reasonable though, and does have the benefit that it's easily typed in to the REPL

**Comment by jystic** [comment]
:hotsprings:

File: doc/internals/todo.md (hunk 3 above)
Comment (tranma) [inline doc/internals/todo.md]: I spied an ML-er! :cake:

Comment (amosr) [inline doc/internals/todo.md]: maybe the "->" should be "=>" just for consistency?

Comment (tranma) [inline doc/internals/todo.md]: :bob: :harper: approves

**Pull request merged**